File: BotTemplate/BotTemplate-Java/src/main/java/io/github/freya022/bot/Bot.java
package io.github.freya022.bot;

import com.freya02.botcommands.api.core.JDAService;
import com.freya02.botcommands.api.core.events.BReadyEvent;
import com.freya02.botcommands.api.core.service.annotations.BService;
import io.github.freya022.bot.config.Config;
import net.dv8tion.jda.api.JDABuilder;
import net.dv8tion.jda.api.entities.Activity;
import net.dv8tion.jda.api.hooks.IEventManager;
import net.dv8tion.jda.api.requests.GatewayIntent;
import org.jetbrains.annotations.NotNull;

import java.util.Set;

@BService
public class Bot extends JDAService {
    private final Config config;

    public Bot(Config config) {
        this.config = config;
    }

    @NotNull
    @Override
    public Set<GatewayIntent> getIntents() {
        return getDefaultIntents();
    }

    @Override
    public void createJDA(@NotNull BReadyEvent event, @NotNull IEventManager eventManager) {
        JDABuilder.createLight(config.getToken(), getIntents())
                .setMaxReconnectDelay(120) //Try to reconnect every 2 minutes instead of the default 15 minutes
                .setActivity(Activity.playing("with Java"))
                .setEventManager(eventManager) //Mandatory
                .build();
    }
}

File: BotTemplate/BotTemplate-Kotlin/src/main/kotlin/io/github/freya022/bot/Bot.kt
package io.github.freya022.bot

import com.freya02.botcommands.api.core.JDAService
import com.freya02.botcommands.api.core.events.BReadyEvent
import com.freya02.botcommands.api.core.service.annotations.BService
import dev.minn.jda.ktx.jdabuilder.light
import io.github.freya022.bot.config.Config
import net.dv8tion.jda.api.entities.Activity
import net.dv8tion.jda.api.hooks.IEventManager
import net.dv8tion.jda.api.requests.GatewayIntent

@BService
class Bot(private val config: Config) : JDAService() {
    override val intents: Set<GatewayIntent> = defaultIntents

    override fun createJDA(event: BReadyEvent, eventManager: IEventManager) {
        // You MUST disable enableCoroutines and set the event manager to the injected one
        light(config.token, intents = intents, enableCoroutines = false) {
            setMaxReconnectDelay(120) //Try to reconnect every 2 minutes instead of the default 15 minutes
            setActivity(Activity.playing("with Kotlin"))
            setEventManager(eventManager)
        }
    }
}